### PR TITLE
[CIT-449] Improve `json_sm`

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

* Add an helper method to add updates to a request without grouping pre-requisite
* Make the type of a module optional

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: sm/json_sm/src/software.rs [hunk 1]
```diff
@@ -6,12 +6,53 @@ pub type SoftwareVersion = String;
...
+            module_type,
+            name,
+            version,
+            url,
+        };
+        module.normalize();
+        module
+    }
+
+    pub fn normalize(&mut self) {
```
Comment (makr11st) [inline sm/json_sm/src/software.rs:41]: Does this need to be a `pub` method?

**Review by makr11st — APPROVED** [review]
Looks good.

**Review by rina23q — APPROVED** [review]
Yay, this PR contains all I need ;)

File: sm/json_sm/src/lib.rs [hunk 2]
```diff
@@ -293,6 +293,146 @@ mod tests {
...
+                    "modules": [
+                        {
+                            "name": "mongodb",
+                            "version": "4.4.6",
+                            "action": "remove"
+                        }
+                    ]
+                }
+            ]
+        }"#;
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply sm/json_sm/src/lib.rs:431]: Yes, that's a good point.

**Pull request merged**